PayPalMessaging/src/main/java/com/braintreepayments/api/PayPalMessagingView.kt

```diff
@@ -0,0 +1,113 @@
+package com.braintreepayments.api
+
+import android.content.Context
+import android.view.ViewGroup
+import com.paypal.messages.PayPalMessageView
+import com.paypal.messages.config.message.PayPalMessageData
+import com.paypal.messages.config.PayPalEnvironment
+import com.paypal.messages.config.message.PayPalMessageConfig
+import com.paypal.messages.config.message.PayPalMessageEventsCallbacks
+import com.paypal.messages.config.message.PayPalMessageStyle
+import com.paypal.messages.config.message.PayPalMessageViewStateCallbacks
+
+/**
+ *  Use [PayPalMessagingView] to display PayPal messages to promote offers such as Pay Later
+ *  and PayPal Credit to customers.
+ * Note: **This module is in beta. It's public API may change or be removed in future releases.**
+ * @property braintreeClient a {@link BraintreeClient}
+ */
+class PayPalMessagingView(
+    private val braintreeClient: BraintreeClient,
+    private val context: Context
+) {
+    var payPalMessagingListener: PayPalMessagingListener? = null
+
+    /**
+     * Creates a view to be displayed to promote offers such as Pay Later and PayPal Credit to customers.
+     * @property request An optional [PayPalMessagingRequest]
+     * Note: **This module is in beta. It's public API may change or be removed in future releases.**
+     */
+    fun start(request: PayPalMessagingRequest = PayPalMessagingRequest()) {
+        braintreeClient.getConfiguration { configuration, configError ->
+            if (configError != null) {
+                payPalMessagingListener?.onFailure(configError)
+            } else if (configuration != null) {
+                val clientId = configuration.payPalClientId
+                if (clientId == null) {
+                    val exception = BraintreeException(
+                        "Could not find PayPal client ID in Braintree configuration."
+                    )
+                    payPalMessagingListener?.onFailure(exception)
+                } else {
+                    val payPalMessageView = constructPayPalMessageView(clientId, configuration, request)
+                    payPalMessageView.layoutParams = ViewGroup.LayoutParams(
+                        ViewGroup.LayoutParams.MATCH_PARENT,
+                        ViewGroup.LayoutParams.MATCH_PARENT
+                    )
+                    payPalMessageView
```

Comment:
Should this have a method called on it? Can't tell if this is a typo or not

Reply:
Nope typo 🤦 removed here: https://github.com/braintree/braintree_android/pull/895/commits/ab4e382821465776d2b881f9853c36870b263740